src/Renderer/Color.cpp

```diff
@@ -48,6 +48,40 @@ const Color Color::NoAlpha(0, 0, 0, 0);
 Color::Color(uint8_t r, uint8_t g, uint8_t b, uint8_t a) : mR(r), mG(g), mB(b), mA(a)
 {}
 
+Color::Color(uint32_t value)
+{
+	setRgbaFromRawValue(value);
+}
+
+void Color::setRgbaFromRawValue(uint32_t value)
+{
+	mR = static_cast<uint8_t>((value & 0xFF000000u) >> 24);
+	mG = static_cast<uint8_t>((value & 0x00FF0000u) >> 16);
+	mB = static_cast<uint8_t>((value & 0x0000FF00u) >> 8);
+	mA = static_cast<uint8_t>((value & 0x000000FFu) >> 0);
+}
+
+void Color::setRgbFromRawValue(uint32_t value)
+{
+	mR = static_cast<uint8_t>((value & 0xFF000000u) >> 24);
+	mG = static_cast<uint8_t>((value & 0x00FF0000u) >> 16);
+	mB = static_cast<uint8_t>((value & 0x0000FF00u) >> 8);
+}
```

Comment:
I was just thinking, this leaves the alpha channel unset. I think for this purpose it would make more sense if the alpha channel was set to `255`.

Reply:
It's intentional. The function name even says so. The alpha value defaults to 255 anyway via the class initializer. This makes raw values that are only 24-bit RGB values from older standards easier to convert to 32-bit by: `setRgbFromRawValue(raw24bit);` (you can then set the alpha value to something other than 255); or, given an external 32-bit value that you want to ignore the alpha value entirely: `setRgbFromRawValue(raw32bit)`.

If you want to set all channels at once, use the `setRgbaFromRawValue` version.

Reply:
This isn't a constructor though. The alpha channel could be set to anything at the point when this function is called.

Actually, I don't think we need a set function. An initialization function should be enough. I think we should avoid modifiers.